pandas/core/internals/managers.py

```diff
@@ -997,6 +997,8 @@ def _make_na_block(
             dtype = interleaved_dtype([blk.dtype for blk in self.blocks])
             if dtype is not None and np.issubdtype(dtype.type, np.floating):
                 fill_value = dtype.type(fill_value)
+            if dtype is not None and np.issubdtype(dtype.type, np.datetime64):
+                fill_value = np.datetime64("NaT")
```

Comment:
can you also pass an appropriate unit to np.datetime64

Reply:
Thanks for the feedback! I was wondering what do you mean by unit?